src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -1260,17 +1269,10 @@ static enum e_op_result execute_stream_op(struct op_and_state* oas) {
           oas->state.state_op_done[OP_RECV_MESSAGE] = true;
 
           /* Extra read to trigger on_succeed */
-          stream_state->rs.read_buffer = stream_state->rs.grpc_header_bytes;
-          stream_state->rs.remaining_bytes = GRPC_HEADER_SIZE_IN_BYTES;
-          stream_state->rs.received_bytes = 0;
-          stream_state->rs.compressed = false;
           stream_state->rs.length_field_received = false;
```

Comment:
I know that this is pre-existing code, but can you comment why these fields (state_op_done and length_field_received) are set in some cases but not others? If it's safe to put it in all cases, consider doing so and relocating that content to the `do_an_extra_read` function for consistency.

Reply:
I chatted with Muxi last week about this, we were not sure why **state_op_done** and **length_field_received** were set in some cases but not others. We thought it would be safe not put these two fields into the `do_an_extra_read` function before figuring out the purpose of this code.

Reply:
`state_op_done` is set when a gRPC message is completely read. When pre-reading the gRPC header (5 bytes), the field should be left untouched. `length_field_received` is set when the gRPC header is completely read, and reset when the message is completely read. They are not related to the "read" action; that's why I don't think they should not belong to the `do_an_extra_read` function.